doc/source/man.rst

```diff
@@ -32,206 +32,217 @@ rho performs four major tasks:
+Use the ``rho auth`` command to create and manage authentication profiles.
 
-There can be multiple auth credentials contained in a single profile.
+An authentication profile defines a set of user credentials to be used during a scan. These user credentials include a username and a password or SSH key. Rho uses SSH to connect to servers on the network and uses authentication profiles to obtain the user credentials that are required access those servers.
 
-Creating and Editing Authentication Credentials
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+When a scan runs, it uses a network profile that contains the hosts or IP addresses to be accessed. The network profile also contains references to the authentication profiles that are required to access those systems. A single network profile can contain a reference to multiple authentication profiles as needed to connect to all systems in that network.
 
-rho uses SSH credentials to access the servers to get their system information during discovery. These credentials can be either a username-password or username-key pair. Each set of credentials is stored in a separate entry.
+Creating and Editing Authentication Profiles
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-**rho auth add --name=** *name* **--username=** *username* **[--password]** **[--sshkeyfile=** *key_file* **]** **[--sudo-password]** **[--vault=** *vault_file* **]**
+To create an authentication profile, supply SSH credentials as either a username-password pair or a username-key pair. Rho stores each set of credentials in a separate authentication profile entry.
 
-``--name=name``
+**rho auth add --name=** *name* **--username=** *username* **(--password | --sshkeyfile=** *key_file* **)** **[--sudo-password]** **[--vault=** *vault_file* **]**
 
-  This required argument sets the name of the new authentication credentials entry. This should be descriptive, such as identifying the user or server it relates to. For example, ``"server1-rhouser"``. It should never contain the actual password, as this name may be logged or printed during rho execution.
+``--name=name``
 
+  Required. Sets the name of the new authentication profile. Use a descriptive name for the value that is meaningful to your organization. For example, you could identify the user or server that the authentication profile relates to, for example, ``admin12`` or ``server1_jdoe``. Do not include the password as part of the name value, because this name might be logged or printed during rho execution.
 
 ``--username=username``
 
-  This required argument contains the username of the SSH identity will use to bind to the server.
+  Required. Sets the username of the SSH identity that is used to bind to the server.
 
 ``--password``
 
-  The argument prompts for the password for the --username identity.
+  Prompts for the password for the ``--username`` identity. Mutually exclusive with the ``--sshkeyfile`` option.
```

Comment:
Maybe adding which takes precedence if both password and sshkey are defined?

The user can set both:

```console
$ rho auth add --name test --password --sshkeyfile ~/.ssh/id_rsa
Please enter your rho vault password: 
Provide connection password.
Password: 
Auth "test" was added
```

Reply:
@elyezer @chambridge @noahl 

Sorry, but from the comment here I can't tell which actually *takes* precedence, and neither can the user. I might suggest that a code or messaging improvement is needed here.
1. Both --password and --sshkeyfile are used in the command. Is there an error message missing at this step to stop the user immediately from causing themselves a problem? e.g., *Use the --password option or the --sshkeyfile option, but not both options.*
2. The output of the command includes a prompt for a "connection password."
3. After puzzling about it for a bit, the user might conclude that the connection password is the value for --password. So they enter it.
4. There's nothing in the messaging here that tells the user which of these, the --password value or the --sshkeyfile value, or both, *was added to the authentication profile.* (So I'm not sure whether your reference to "precedence" means one is discarded.) So, if messaging does not exist for item 1, then here's a different message opportunity, e.g., *You entered a value for both the --password option and the --sshkeyfile option. These values are mutually exclusive. Only one of these values, the X value, was added to the authentication profile.*
5. There's nothing in the messaging here that *tells the user which one took precedence--* regardless of whether action was that only one was added to the profile, or both were added to the profile but only one is actually getting used when the profile is called. So let's say messaging for item 1 or item 4 doesn't exist. Then here's the message opportunity: *You entered a value for both the --password option and the --sshkeyfile option. These values are mutually exclusive. Although both values were added to the authentication profile, only the Y value will be used.*

Reply:
In the meantime, if someone tells me what happens to the values and which takes precedence in the existing code, then I can add a brief statement about it, similar to the messaging in the previous comment.

Reply:
Its a bug if you can do both options. You should only be able to do one or the other.

Reply:
An issue needs to be created for it then. Showing the auth I've created on the first comment I see the following:

```console
$ rho auth show --name test
Please enter your rho vault password: 
{
    "id": "1a764147-91fc-41cd-bf17-7c480235d343",
    "name": "test",
    "password": "********",
    "ssh_key_file": "/home/elyezer/.ssh/id_rsa",
    "sudo_password": null,
    "username": null
}
```

Reply:
https://github.com/quipucords/rho/pull/331 will get this fixed. Thank you @chambridge.